torchdata/datapipes/iter/util/dataframemaker.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+from functools import partial
+from typing import List, Optional, TypeVar
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+try:
+    import torcharrow
+except ImportError:
+    torcharrow = None
+
+T_co = TypeVar("T_co")
+
+
+@functional_datapipe("convert_to_dataframe")
+class DataFrameMakerIterDataPipe(IterDataPipe):  # IterDataPipe[torcharrow.IDataFrame[T_co]]
+    r"""
+    Iterable DataPipe that takes rows of data, batch a number of them together and create TorchArrow DataFrames.
+    Note that there is a trade-off between having a large number of rows within a DataFrame and usage of memory.
+
+    Args:
+        source_dp: IterDataPipe containing rows of data
+        dataframe_size: number of rows of data within each DataFrame
+        dtype: specify the TorchArrow dtype for the DataFrame
+        columns: List of str that specifies the column names of the DataFrame
+        device: specify the device on which the DataFrame will be stored
+    """
+
+    def __new__(
+        cls,
+        source_dp: IterDataPipe[T_co],
+        dataframe_size: int = 1000,  # or Page Size
+        dtype=None,  # Optional[torcharrow.dtypes.DType]
+        columns: Optional[List[str]] = None,
+        device: str = "",
+    ):
+        # In this version, DF tracing is not available, which would allow DataPipe to run DataFrame operations
+        batch_dp = source_dp.batch(dataframe_size)
+        df_dp = batch_dp.map(partial(torcharrow.DataFrame, dtype=dtype, columns=columns, device=device))
+        return df_dp
```

Comment:
Since we have two different `csvparser` and one is `CSVDictParser` , could you please validate if we can emit `columns` argument to construct dataframe?

https://github.com/pytorch/data/blob/1ec936b8954f9a525e5e2c2363708cdc388eb5d2/torchdata/datapipes/iter/util/plain_text_reader.py#L187

Reply:
I just expanded on one of the test examples we have, and it works as expected.
```python
csv_files = {"1.csv": "key,item\na,1\nb,2"}
self._custom_files_set_up(csv_files)
datapipe1 = FileLister(self.temp_dir.name, "*.csv")
datapipe2 = FileOpener(datapipe1, mode="b")
datapipe3 = datapipe2.map(get_name)
csv_dict_parser_dp = datapipe3.parse_csv_as_dict()

import torcharrow
import torcharrow.dtypes as dt
DTYPE = dt.Struct([dt.Field("key", dt.string), dt.Field("item", dt.string)])
print("\nDataFrame")
print(list(csv_dict_parser_dp.dataframe(dtype=DTYPE, columns=['key', 'item'])))
```

Reply:
I will add it as a test case